test/config-luatest/cluster_config_test.lua

```diff
@@ -311,3 +311,28 @@ g.test_misplace_option = function(g)
     cluster.startup_error(g, config, "replicaset \"sharding\" should " ..
                                      "include at least one instance.")
 end
+
+g.test_groups_with_same_name = function(g)
+    -- We aren't able to use cbuilder helper here since it
+    -- internally uses cluster_config not allowing configs
+    -- with the same group/replicaset/instance names.
+    local config = [[
+        iproto:
+          listen:
+            - uri: unix/:./{{ instance_name }}.iproto
+        groups:
+          g-001:
+            replicasets:
+              r-001:
+                instances:
+                  i-001: {}
+          g-001:
+            replicasets:
+              r-002:
+                instances:
+                  i-002: {}
+    ]]
+
+    cluster.startup_error(g, config, 'found groups with the same ' ..
+                                     'name "g-001".')
+end
```

Comment:
The `startup_error` helper unconditionally encodes the passed config value as YAML. Seems like we have double encoding here.

Reply:
I understood why the test cases pass: `instance_names_from_config` returns an empty table and no checks are performed.

Reply:
Thank you for your great attention during reviewing.

```diff
--- a/test/config-luatest/cluster.lua
+++ b/test/config-luatest/cluster.lua
@@ -260,6 +260,7 @@ end
 -- given error message.
 local function startup_error(g, config, exp_err)
     assert(g)  -- temporary stub to not fail luacheck due to unused var
+    assert(type(config) == 'table')
     assert(config._config == nil, "Please provide cbuilder:new():config()")
     -- Prepare a temporary directory and write a configuration
     -- file.
```

I've added the assertion in the cluster helper. I hope it's ok that such a small change is a part of the first patch. I suppose it'll help to avoid such mistakes.